### ISSUE TITLE
Make documentation decent if a term is used more than once.

### ISSUE THREAD

**Issue by dbuenzli** [dbuenzli/cmdliner#77]
Now that I have tried to unmess the internals a bit I think that what @drup says in [this comment](https://github.com/dbuenzli/cmdliner/pull/9#issuecomment-128010442) is actually easy to achieve. 

I think that the first point he mentions is already valid. The second one should be easy by simply turning [this list](https://github.com/dbuenzli/cmdliner/blob/0c5806f72bfa4eb73583bff6c1bf7fd2dfaf807d/src/cmdliner_docgen.ml#L179) into a set based on the [argument id](https://github.com/dbuenzli/cmdliner/blob/0c5806f72bfa4eb73583bff6c1bf7fd2dfaf807d/src/cmdliner_info.ml#L41).

**Comment by dbuenzli** [comment]
Note that this not based on comparing option *names*. If two persons define the same option name with different docs (i.e. two different `Arg.info`s) and they are used in the same program then it will appear more than once in the docs.

**Comment by dbuenzli** [comment]
So something is in with 9bedab3 in the end I changed to collect sets of args (vs lists before) directly in the applicative functor. 

Regarding:

> If two persons define the same option name with different docs (i.e. two different Arg.infos)

The mentioned commit has an example that does this however one of the dupes will never get parsed. I guess this is because it gets replaced during the parsing trie construction [here](https://github.com/dbuenzli/cmdliner/blob/9bedab3543713b59c72a021e3aa9b5a3141c80c8/src/cmdliner_cline.ml#L37-L40) that maps option names to argument information. 

We could map names to list of argument information but in general this seems a bit dubious if the same option is defined in a conflicting manner (e.g. with a value of different type) then a user will never be able to use the option (one of the option value parser will always fail), so I think I'm going to detect replacements in the trie and failwith `Invalid_argument` in that case. 

So basically a term is not allowed to have one option name defined via two different `Arg.info`s, it this happens [Invalid_argument] is raised.

**Comment by dbuenzli** [comment]
1c7c9ed disallows multiple option or command name definitions.
